compiler/rustc_infer/src/traits/util.rs

```diff
@@ -235,6 +236,37 @@ pub fn elaborate<'tcx, O: Elaboratable<'tcx>>(
     elaborator
 }
 
+pub fn elaborate_predicates_of<'tcx>(
```

Comment:
Is this only guaranteed to be called on a trait? Never a trait item? If so, using this function is a hazard because it only computes the predicates of the item and not its parents. I would prefer if you remove it and inline it into its callsites like:

```
elaborate(tcx, tcx.predicates_of(def_id).instantiate_own_identity())
```

Though if this *is* intended to be called on non-top-level-items, you need to use:

```
elaborate(tcx, tcx.predicates_of(def_id).instantiate_identity())
```

(and a `.map(|(clause, sp)| (clause.as_predicate(), sp))` if necessary, but you can likely rework the call-sites to operate solely on `Clause`s and not `Predicate`s.)